repositories/answer-sheets.ts

```diff
@@ -0,0 +1,144 @@
+import { Database } from "@/models/supaservice.types";
+import { SupabaseClient } from "@supabase/supabase-js";
+
+export class AnswerSheetRepository {
+  private db: SupabaseClient<Database>;
+
+  constructor(supabase: SupabaseClient<Database>) {
+    this.db = supabase;
+  }
+
+  resetAnswerSheetConfig = async (
+    answerSheetId: string,
+    data: {
+      counts: number;
+      nOptions?: number;
+      points?: number;
+    }
+  ) => {
+    const { counts, nOptions, points } = data;
+
+    const { data: existingData, error: existingError } = await this.db
+      .from("answer_sheets")
+      .select("*")
+      .eq("uuid", answerSheetId)
+      .single();
+
+    if (existingError || !existingData)
+      throw { message: "Data tidak ditemukan" };
+    const {
+      counts: oldCounts,
+      n_options: oldNOptions,
+      points: oldPoints,
+    } = existingData;
+
+    const new_n_options =
+      nOptions && oldCounts >= counts
+        ? Array(counts).fill(nOptions)
+        : counts > oldCounts
+        ? Array(counts).fill(nOptions ?? oldNOptions[0])
+        : undefined;
+    const new_points =
+      points && oldCounts >= counts
+        ? Array(counts).fill(points)
+        : counts > oldCounts
+        ? Array(counts).fill(points ?? oldPoints[0])
+        : undefined;
+
+    // update answer_sheets
+    const { error: updateError } = await this.db
+      .from("answer_sheets")
+      .update({
+        counts,
+        n_options: new_n_options,
+        points: new_points,
+      })
+      .eq("uuid", answerSheetId);
+
+    if (updateError)
+      throw { message: "Gagal mengupdate data", error: updateError };
+  };
```

Comment:
With the current tech stack utilizing Supabase as our BaaS, transactions are not natively supported. To implement a transaction, we would need to create a PostgreSQL function and invoke it via supabase.rpc(). However, this approach would offload the logic from our codebase to the Supabase configuration, which could result in less control and make maintenance more challenging.

Do you think it would be a better approach to offload this logic despite the trade-offs in control and maintainability?